### ISSUE TITLE
[expo-av][android] Audio Playback not working on Android (Works on iOS)

### ISSUE THREAD

**Issue by archcorsair** [expo/expo#20165]
### Summary

# Error
```
[Error: java.lang.IllegalStateException: Content of this type is unsupported at the moment. Unsupported type: 4]
```
# Summary
This snack is the basic example from the expo-av docs: https://snack.expo.dev/@archcorsair/playing-sounds
It works beautifully on android and I get audio playback, however with the same exact code within my own project I get the error listed above. Is this a known issue for the version combination I am using?

- Testing with an MP3 file that has worked with `expo-av` in the past and also with the mp3 file that is provided in the expo snack (https://snack.expo.dev/@archcorsair/playing-sounds) yielded same results.

- Tried both local and remote resource for audio

# Expected Behavior
Expect audio playback to work on android

# Actual
Only works on iOS

# Additional Notes
Android: 13 Google Pixel 6 Pro

Another user experiencing the exact issue:
https://forums.expo.dev/t/expo-av-audio-failed-to-load-error-error-java-lang-illegalstateexception-content-of-this-type-is-unsupported-at-the-moment-unsupported-type-4

### What platform(s) does this occur on?

Android

### Environment

`"expo-av": "^12.0.4"`

```
expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 13.0.1
      Shell: 5.8.1 - /bin/zsh
    Binaries:
      Node: 16.17.1 - ~/Library/Caches/fnm_multishells/2193_1668730273410/bin/node
      Yarn: 1.22.19 - /opt/homebrew/bin/yarn
      npm: 8.15.0 - ~/Library/Caches/fnm_multishells/2193_1668730273410/bin/npm
      Watchman: 2022.11.14.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.11.3 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 21.4, iOS 16.0, macOS 12.3, tvOS 16.0, watchOS 9.0
      Android SDK:
        API Levels: 27, 28, 30, 31, 32, 33
        Build Tools: 30.0.2, 30.0.3, 31.0.0, 32.0.0, 33.0.0
        System Images: android-31 | Google APIs ARM 64 v8a
    IDEs:
      Android Studio: 2021.3 AI-213.7172.25.2113.9123335
      Xcode: 14.0.1/14A400 - /usr/bin/xcodebuild
    npmPackages:
      babel-preset-expo: ^9.2.0 => 9.2.2
      expo: ^46.0.17 => 46.0.17
      react: ^18.0.0 => 18.2.0
      react-dom: 17.0.1 => 17.0.1
      react-native: ^0.69.5 => 0.69.7
      react-native-web: 0.17.1 => 0.17.1
    Expo Workflow: bare
 ```

### Minimal reproducible example

I would like to preface that this super bare snack works on my android device, but my own project with the SAME code does not:
https://snack.expo.dev/@archcorsair/playing-sounds

**Comment by brentvatne** [comment]
@archcorsair - happy to investigate if you can share a project that we can clone that reproduces the issue. if the snack is insufficient then a github repo would be good

**Comment by archcorsair** [comment]
@brentvatne thanks!
It will be exceedingly difficult for me to share my repo, as its private, and would probably result in my immediate termination 😆 However, I'm happy to provide as much possible context as I can, including all related packages that are installed and their versions.
I would like to also reiterate that the code snippet from the snack was tested during my debugging and it also did not work: same error.

I noticed this issue after upgrading from expo 43 to 46 and RN from 0.64 to 0.69

### List of deps
```json
  "dependencies": {
    "@adobe/react-native-acpanalytics": "^2.0.1",
    "@adobe/react-native-acpcore": "^2.0.1",
    "@adobe/react-native-acpuserprofile": "^2.0.1",
    "@adobe/react-native-aepassurance": "^2.0.1",
    "@expo/vector-icons": "^12.0.0",
    "@notifee/react-native": "^7.0.1",
    "@react-native-async-storage/async-storage": "^1.17.10",
    "@react-native-community/blur": "^4.1.0",
    "@react-native-community/cli": "^8.0.6",
    "@react-native-community/slider": "^4.3.1",
    "@react-native-firebase/analytics": "^14.12.0",
    "@react-native-firebase/app": "^14.12.0",
    "@react-native-firebase/crashlytics": "^14.12.0",
    "@react-native-firebase/messaging": "^14.12.0",
    "@react-native-masked-view/masked-view": "^0.2.7",
    "@react-navigation/bottom-tabs": "^5.11.9",
    "@react-navigation/devtools": "^5.1.22",
    "@react-navigation/material-top-tabs": "^5.3.15",
    "@react-navigation/native": "^5.9.4",
    "@react-navigation/stack": "^5.14.4",
    "@sparkfabrik/react-native-idfa-aaid": "^0.8.3",
    "apptentive-react-native": "5.8.3",
    "date-time-format-timezone": "^1.0.22",
    "deprecated-react-native-prop-types": "^2.3.0",
    "expo": "^46.0.10",
    "expo-application": "^4.2.2",
    "expo-asset": "^8.6.1",
    "expo-av": "^12.0.4",
    "expo-constants": "^13.2.4",
    "expo-font": "^10.2.0",
    "expo-keep-awake": "^10.2.0",
    "expo-linear-gradient": "^11.4.0",
    "expo-status-bar": "^1.4.0",
    "expo-web-browser": "^11.0.0",
    "htmlparser2": "^7.2.0",
    "intl": "^1.2.5",
    "lodash": "^4.17.21",
    "lottie-ios": "^3.4.0",
    "lottie-react-native": "^5.1.4",
    "metro-config": "^0.70.3",
    "moti": "^0.10.1",
    "prop-types": "^15.7.2",
    "react": "^18.0.0",
    "react-dom": "17.0.1",
    "react-native": "^0.69.5",
    "react-native-autoheight-webview": "^1.6.4",
    "react-native-dash": "0.0.11",
    "react-native-device-info": "^10.0.2",
    "react-native-geocoding": "^0.5.0",
    "react-native-gesture-handler": "~1.10.2",
    "react-native-google-mobile-ads": "^4.1.0",
    "react-native-htmlview": "^0.16.0",
    "react-native-jw-media-player": "^0.1.57",
    "react-native-lottie-splash-screen": "^1.0.1",
    "react-native-maps": "^1.3.1",
    "react-native-orientation-locker": "^1.4.0",
    "react-native-reanimated": "^2.10.0",
    "react-native-redash": "^16.2.3",
    "react-native-restart": "^0.0.22",
    "react-native-safe-area-context": "3.3.2",
    "react-native-screens": "^2.18.1",
    "react-native-snap-carousel": "^3.9.1",
    "react-native-svg": "12.1.1",
    "react-native-svg-transformer": "^0.14.3",
    "react-native-tab-view": "^2.15.1",
    "react-native-tracking-transparency": "^0.1.1",
    "react-native-watch-connectivity": "^1.0.4",
    "react-native-web": "0.17.1",
    "react-native-webview": "^11.3.2",
    "react-navigation-header-buttons": "^4.0.2",
    "react-redux": "^7.2.0",
    "redux": "^4.0.5",
    "redux-logger": "^3.0.6",
    "redux-thunk": "^2.3.0"
  },
```

**Comment by brentvatne** [comment]
@archcorsair - from the above comment:

> "I've only been able to reproduce it in private, proprietary code"
You may not have spent enough time narrowing down the root cause of the issue. Try out the techniques discussed in this [manual debugging guide](https://expo.fyi/manual-debugging) to learn how to isolate the problem from the rest of your codebase.

**Comment by archcorsair** [comment]
@brentvatne I appreciate the resource, thanks for that!

I spent a big chunk of time debugging this by isolating the component we use away from the business logic, which eventually lead me to just copy/paste the expo-av example and use that as the component.

I narrowed it down simply to: different `expo-av` versions. I provided the deps in case there is a scenario where `expo-av` might not play nice with RN or expo for specific versions. Prior to upgrading those audio playback was working just fine.

If I run the expo snack using `expo-av` 13 it works just fine, however using that version is currently not possible in our build so this is not an option at this moment.
